front/src/utils/helpers.ts

```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect } from 'react';
+import 'moment/locale/fr';
+
+/**
+ * Fonctions utiles pour les cartes des différents projets
+ */
+
+/**
+ * Debounce input fields
+ */
+export const useDebounce = (value: string | number, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+  return debouncedValue;
+};
+
+/**
+ * Debounce input fields
+ */
+export const useDebouncedFunc = (
+  value: string | number,
+  delay: number,
+  func: (newValue: number) => void
+) => {
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      func(value as number);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+};
+
+/**
+ * Donne le bon type de curseur au survol
+ * @param {boolean} isHovering - S'il est à True, retourne le bon type de pointeur
+ * @returns {string} La classe correspondant au pointeur
+ */
+export function getCursor(isHovering: boolean) {
+  return isHovering ? 'pointer' : 'default';
+}
+
+/**
+ * Donne la propriété en string de la visibilité d'un layer
+ * @param {boolean} isVisible - Si le layer doit être visible ou non
+ * @returns 'visible' si le layer doit être visible, 'none' sinon
+ */
+export function booleanToStringLayerVisibility(isVisible: boolean) {
+  return isVisible ? 'visible' : 'none';
+}
+
+/**
+ * Donne la valeur arrondie d'une coordonnée GPS
+ * @param {number} val - La valeur à arrondir
+ * @returns {number} La coordonnée arrondie au 10 000ème
+ */
+export function gpsRound(val: number) {
+  return Math.round(val * 10000) / 10000;
+}
+
+/**
+ * Fonction de callback lors d'un requête vers une URL extérieure
+ * @param url - L'URL de la requête
+ * @param resourceType - Le type de ressource
+ * @param urlmap - L'URL de base pour la carte
+ * @returns Le token d'authentifcation lorsqu'il est nécessaire
+ */
+export function transformRequest(url: string, resourceType: string, urlmap: string) {
```

Comment:
ResourceType from Maplibre and also rename the function
```suggestion
export function transformMapRequest(url: string, resourceType: ResourceType | undefined, mapUrl: string) {
```

Reply:
done